File: 27.移除元素.go
/*
 * @lc app=leetcode.cn id=27 lang=golang
 *
 * [27] 移除元素
 */

// @lc code=start
func removeElement(nums []int, val int) int {
	// for i := len(nums) - 1; i >= 0; i-- {
	// 	if nums[i] == val {
	// 		nums = append(nums[:i], nums[i+1:]...)
	// 	}
	// }
	// return len(nums)
	i:= 0
	for _, v := range nums {
		if v != val {
			nums[i] = v
			i++
		}
	}
	return len(nums[:i])
}

// @lc code=end

File: 27.移除元素.py
#
# @lc app=leetcode.cn id=27 lang=python3
#
# [27] 移除元素
#

# @lc code=start
import enum


class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        i = 0
        for _, v in enumerate(nums):
            if v != val:
                nums[i] = v
                i = i + 1
        return i

# @lc code=end
